slack-base/src/main/java/com/hubspot/slack/client/methods/params/chat/MessageParams.java

```diff
@@ -1,13 +1,22 @@
 package com.hubspot.slack.client.methods.params.chat;
 
+import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
 
+import org.immutables.value.Value;
+
 import com.hubspot.slack.client.methods.interceptor.HasChannel;
 import com.hubspot.slack.client.models.Attachment;
+import com.hubspot.slack.client.models.blocks.Block;
 
 public interface MessageParams extends HasChannel {
   Optional<String> getText();
 
   List<Attachment> getAttachments();
+
+  @Value.Default
+  default List<Block> getBlocks() {
```

Comment:
Per my comment, you don't need this. Immutables default collections to the empty collection if they're not set, so you actually get this behavior without needing the explicit description of it here.

Reply:
Keith has suggested adding this in would protect against anyone that might (who knows!) be implementing or extending `MessageParams` themselves.  I know it's not necessary for the immutables, but doing it this way makes this a non-breaking change.  What are your thoughts on this @szabowexler ?

Reply:
Oh I see what you're saying. I think that's a violation of the contract here...this is really just intended as a way to make our annotation processor handle it. I don't really think we should be allowing people to extend it in arbitrary ways

Reply:
It's a public interface though. If it was just meant for internal immutables/annotation processing only then it probably shouldn't have been made public. As a safety measure I think the default implementation is the only way to not piss off users of this library